scripts/http-jsonp-detection.nse

```diff
@@ -0,0 +1,174 @@
+local http = require "http"
+local shortport = require "shortport"
+local stdnse = require "stdnse"
+local string = require "string"
+local json = require "json"
+local url = require "url"
+local httpspider = require "httpspider"
+
+description = [[
+Attempts to discover JSONP endpoints in web servers. JSONP endpoints can be
+used to bypass Same-origin Policy restrictions in web browsers.
+
+The script searches for callback functions in the response to detect JSONP
+endpoints. It also tries to determine callback function through URL(callback 
+function may be fully or partially controllable from URL) and also tries to 
+bruteforce the most common callback variables through the URL.
+
+References : https://securitycafe.ro/2017/01/18/practical-jsonp-injection/
+
+]]
+
+---
+-- @usage
+-- nmap -p 80 --script http-jsonp-detection <target>
+--
+-- @output
+-- 80/tcp open  http    syn-ack
+-- | http-jsonp-detection: 
+-- |   JSONP endpoint found at /rest/contactsjp.php. Function name is callback
+-- |_  Callback function is completely controllable from the URL
+-- 
+--
+-- @xmloutput
+-- <table>
+-- <elem>JSONP endpoint found at /rest/contactsjp.php. Function name is callback</elem>
+-- <elem>Callback function is completely controllable from the URL</elem>
+-- </table>
+-- 
+-- @args http-jsonp-detection.path The URL path to request. The default path is "/".
+--- 
+
+author = {"Vinamra Bhatia"}
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {""} --to be figured out!
```

Comment:
auth:Does the script do anything related to auth? No
broadcast:Does the script send broadcast requests? No
brute:Does the script brute force a service? No. 
discovery:Does the script discovers additional information about the target? Yes.
dos: Is this a DoS? No.
exploit: Is this an exploit? No.
external: Does it depend on an external source? No.
fuzzer:Does it fuzz? No.
intrusive: Is it intrusive? No.
malware: Is it related to malware? No.
safe> Is it safe to run agasint web servers? Yes.
version: Does it set version information? No.
vuln: Is it related to a vulnerability check? Yes.